MatrixSDK/Data/RoomList/Common/MXSuggestedRoomListDataFetcher.swift

```diff
@@ -88,10 +94,16 @@ internal class MXSuggestedRoomListDataFetcher: NSObject, MXRoomListDataFetcher {
             }
             self.refresh()
         })
+        sessionDidSyncObserver = NotificationCenter.default.addObserver(forName: NSNotification.Name.mxSessionDidSync, object: nil, queue: OperationQueue.main) { [weak self] notification in
+            self?.updateData()
+        }
     }
     
     private func removeDataObservers(for space: MXSpace?) {
         space?.room?.removeListener(spaceEventsListener)
+        if let observer = sessionDidSyncObserver {
```

Comment:
If we're removing the listener in this method, we should add a new one in `addDataObservers(for space:`method. Otherwise, we would lose it when updating the space.

Reply:
Not sure to follow you on this one. `sessionDidSyncObserver` is already set in `addDataObservers(for space: MXSpace?)`

Reply:
Ah sorry, i saw it wrong.